### PR TITLE
Static Application Security Testing using MobSF and Docker

### LINKED ISSUE #6058: [Security] How to perform containerized static application security testing(SAST) using MobSF within a Docker container
#### Proposed title of article
Static application security testing(SAST) using MobSF in a Docker container

#### Proposed article introduction
This article will introduce a reader to application vulnerability testing, outlining the various ways of performing the same. This article will try to make a brief comparison between the static testing method and dynamic method and any other methods in existence.

The article will further focus on the static secutity testing method outlining step by step process of how it works and analyze the results obtained from SAST together with its advantages. This article will use MobSF framework from within a docker container to perform the step by step process of SAST. 

#### Key takeaways 
The reader should master:
1. What is SAST
2. Differences between SAST and DAST
3. Tools used in application security testing
4. Step by Step process of how SAST works
5. Advantages and Disadvantages SAST

#### Article quality
This article will not only cover a detailed process of application security testing but also give the various tools used apart from the tool of focus, and also explain what security testing entails in theory.

#### References
N/A

### PR DESCRIPTION
**Pull request by RuthAMare**
#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

#### Topic suggestion this closes
This closes #6058

### CODE REVIEW

**Comment by srishilesh** [comment]
Hello @RuthAMare ,

Thanks for submitting this PR.

Your PR has entered our reviewing queue. Please wait until a peer-reviewer pings you to get started.

Meanwhile, we suggest you to go through existing approved PRs/published articles to see if you have missed out anything.

Thanks.

**Comment by louisefindlay23** [comment]
Plagiarism check complete

File: content/articles/how-to-perform-containerized-static-application-security-testing(sast)-using-mobsf-within-a-docker-container/index.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+A vulnerability that is not fixed could lead to dire consequences for the user and owners of an application. SAST is one of the ways used to scan and assess applications for vulnerabilities.
+<!--more-->
+
+### Table of content
+- [Prerequisites](#prerequisites)
+- [What is SAST](#what-is-sast)
+- [Differences between SAST and DAST](#differences-between-sast-and-dast)
+- [Tools used to perform SAST](#tools-used-to-perform-sast)
+- [Step by Step process of SAST](#step-by-step-process-of-sast)
+- [Advantages and Disadvantages of SAST](#advantages-and-disadvantages-of-sast)
```
Comment (collins-a) [inline content/articles/how-to-perform-containerized-static-application-security-testing(sast)-using-mobsf-within-a-docker-container/index.md:26]: Fix this link to reflect the subtopics in the body of your tutorial

File: content/articles/how-to-perform-containerized-static-application-security-testing(sast)-using-mobsf-within-a-docker-container/index.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+The above scenario is also known as white-box testing.
+
+Developers can therefore identify shortcomings of an application in the early stages of development and quickly resolve issues that may otherwise affect the final rolled out application.
+
+Developers can systematically track the vulnerability trends within the application thereby promoting the creation of a secure and effecint Software Development Life Cycle(SDLC).
+
+SAST provides visual representations of the problems identified making it easy to navigate the code. In addition, it provides suggestions on how to fix the vulnerabilities identified within the code. 
+
+Customized reports downloadable and visible offline are supported by most SAST tools and can also be tracked through dashboards.
```
Comment (collins-a) [inline content/articles/how-to-perform-containerized-static-application-security-testing(sast)-using-mobsf-within-a-docker-container/index.md:47]: Consider arranging this section into paragraphs as opposed to rather isolated lines

File: content/articles/how-to-perform-containerized-static-application-security-testing(sast)-using-mobsf-within-a-docker-container/index.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+### Differences between SAST and DAST
+There are two different types of Application Security Testing, i.e., SAST and DAST (Dynamic Application Security Testing).
+
+The above-mentioned detect security vulnerabilities in applications except in a unique way for each. Below are some of the critical comparisons of SAST between DAST:
```
Comment (collins-a) [inline content/articles/how-to-perform-containerized-static-application-security-testing(sast)-using-mobsf-within-a-docker-container/index.md:52]: Consider simplifying this line to better convey the contrast between SAST and DAST

File: content/articles/how-to-perform-containerized-static-application-security-testing(sast)-using-mobsf-within-a-docker-container/index.md [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+- SAST supports all types of software while DAST can only scan applications such as web applications and web services.
+- SAST cannot discover runtime and environment-related issues, while DAST can discover runtime and environment-related issues.
+- It costs less to fix vulnerabilities using SAST, while it is relatively expensive to fix them using DAST.
+- SAST discovers vulnerabilities earlier into the SDLC, while DAST discovers vulnerabilities towards the end of SDLC
+- SAST requires source code DAST requires a running application.
+
+### Prerequisites
+The learner should be familiar with:
+1.	Git commands and commands in a Linux operating system.
+2.	A basic understanding of containerized storage.
```
Comment (collins-a) [inline content/articles/how-to-perform-containerized-static-application-security-testing(sast)-using-mobsf-within-a-docker-container/index.md:63]: It is likely that a reader may appreciate having the prerequisites section higher up the article to help them decide whether to continue reading.

**Review by collins-a — CHANGES_REQUESTED** [review]
Decent tutorial on SAST @RuthAMare. However, consider my review comments. Correctly paragraph your sentences where applicable and have a quick Grammarly check to handle a couple of errors.

**Comment by RuthAMare** [comment]
> Decent tutorial on SAST @RuthAMare. However, consider my review comments. Correctly paragraph your sentences where applicable and have a quick Grammarly check to handle a couple of errors.

Thanks @collins-a , all changes made accordingly.

**Review by collins-a — APPROVED** [review]

**Comment by ahmadmardeni1** [comment]
Great work team. @RuthAMare @srishilesh

**Pull request merged**